src/core/shape_inference/include/strided_slice_shape_inference.hpp

```diff
@@ -176,40 +165,24 @@ void shape_infer(const StridedSlice* op,
             // calculating dimension (begin, end, begin_mask, end_mask, stride)
             else if (got_begin && got_end && got_strides) {
                 // set default value for stride or use given value
+                const auto& input_dim = input_shape[input_shape_idx];
                 auto stride = (strides.size() > static_cast<size_t>(axis)) ? strides[axis] : static_cast<int64_t>(1);
                 NODE_VALIDATION_CHECK(op, stride != 0, "Stride must be non-zero");
-                // normalize by add max to value if negative
-                const auto normalize = [](const int64_t& value, const int64_t& max) -> int64_t {
-                    return (value < 0) ? value + max : value;
-                };
-
-                // clip value to min, max
-                const auto clip = [](const int64_t& value, const int64_t& min, const int64_t& max) -> int64_t {
-                    return std::min(std::max(value, min), max);
-                };
 
                 // get stride output dimension for dimension and bounds
                 // may not be called for stride 0 (div by 0!!!) assert check done above
-                const auto get_output_dim = [&](const int64_t& dim, const int64_t& lower, const int64_t& upper) {
+                const auto get_stride_elements = [&](const int64_t& dim, const int64_t& lower, const int64_t& upper) {
```

Comment:
Similar to `get_step_elements`, the `get_stride_elements` name is misleading, the previous name `get_output_dim` was more related to the function body, the returned value and the description above.

Reply:
Function is renemed and code is refactored.